v0/destinations/webengage/transform.js

```diff
@@ -0,0 +1,124 @@
+const { EventType } = require("../../../constants");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  WEBENGAGE_IDENTIFY_EXCLUSION,
+  BASE_URL,
+  BASE_URL_IND
+} = require("./config");
+
+const {
+  defaultRequestConfig,
+  constructPayload,
+  defaultPostRequestConfig,
+  CustomError,
+  ErrorMessage,
+  extractCustomFields,
+  getValueFromMessage
+} = require("../../util");
+const set = require("set-value");
+const moment = require("moment");
+const { isValidTimestamp } = require("../ometria/util");
+
+const responseBuilder = (message, category, { Config }) => {
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  if (!payload.userId && !payload.anonymousId) {
+    throw new CustomError(
+      "[WEBENGAGE]: Either one of userId or anonymousId is mandatory.",
+      400
+    );
+  }
+  let baseUrl, endpoint;
+  if (Config.dataCenter === "ind") {
+    baseUrl = BASE_URL_IND;
+  } else {
+    baseUrl = BASE_URL;
+  }
+
+  if (category.type === "identify") {
+    const customAttributes = {};
+    extractCustomFields(
+      message,
+      customAttributes,
+      ["context.traits", "traits"],
+      WEBENGAGE_IDENTIFY_EXCLUSION
+    );
+
+    payload = { ...payload, attributes: customAttributes };
+    endpoint = `${baseUrl}/${Config.licenseCode}/users`;
+  } else {
+    let eventTimeStamp = message?.properties?.eventTime;
+    let finalTimeStamp;
+    if (isValidTimestamp(eventTimeStamp)) {
+      finalTimeStamp = eventTimeStamp;
+    } else {
+      eventTimeStamp = getValueFromMessage(message, [
+        "timestamp",
+        "originalTimestamp"
+      ]);
+      if (isValidTimestamp(eventTimeStamp)) {
+        finalTimeStamp = eventTimeStamp;
+      }
+    }
+    if (finalTimeStamp) {
+      finalTimeStamp = moment(finalTimeStamp).format("YYYY-MM-DDThh:mm:sZZ");
+      set(payload, "eventTime", finalTimeStamp);
+    }
```

Comment:
You can simplify it.
```suggestion
    let eventTimestamp = getValueFromMessage(message, [
        "properties.eventTime",
        "timestamp",
        "originalTimestamp"
      ]);
    if (isValidTimestamp(eventTimestamp)) {
      finalTimeStamp = moment(eventTimestamp).format("YYYY-MM-DDThh:mm:sZZ");
      set(payload, "eventTime", finalTimeStamp);
    }
```

Reply:
Its updated